pkg/pushsync/pushsync.go

```diff
@@ -147,10 +176,89 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 				return fmt.Errorf("chunk store: %w", err)
 			}
 
+			count := 0
+			// Push the chunk to some peers in the neighborhood in parallel for replication.
+			// Any errors here should NOT impact the rest of the handler.
+			err = ps.topologyDriver.EachNeighbor(func(peer swarm.Address, po uint8) (bool, bool, error) {
+
+				// skip forwarding peer
+				if peer.Equal(p.Address) {
+					return false, false, nil
+				}
+
+				if count == nPeersToPushsync {
+					return true, false, nil
+				}
+				count++
+
+				go func(peer swarm.Address) {
+
+					var err error
+					defer func() {
+						if err != nil {
+							ps.logger.Tracef("pushsync replication: %v", err)
+							ps.metrics.TotalReplicatedError.Inc()
+						} else {
+							ps.metrics.TotalReplicated.Inc()
+						}
+					}()
+
+					// price for neighborhood replication
+					const receiptPrice uint64 = 0
+
+					headers, err := headerutils.MakePricingHeaders(receiptPrice, chunk.Address())
```

Comment:
does this not shadow the `err` in defer func?

Reply:
no